enigma-principal/app/src/controller/boot_network.rs

```diff
@@ -0,0 +1,123 @@
+        pk_32.copy_from_slice(private_key);
+        let signer = Box::new(PrivateKeyEthereumSigner::new(pk_32)) as Box<dyn EcdsaSign + Send + Sync>;
+        signer
+    } else {
+        let signer = Box::new(SgxEthereumSigner::new(eid)) as Box<dyn EcdsaSign + Send + Sync>;
+        signer
+    }
+}
+
+fn get_signing_address(eid: sgx_enclave_id_t) -> Result<String, EnclaveError> {
+    Ok(equote::get_register_signing_address(eid).or(Err(EnclaveError::UnDetailedEnclaveErr))?.to_hex())
+}
+
+fn get_ethereum_address(eid: sgx_enclave_id_t, config: KMConfig) -> Result<String, EnclaveError> {
+    if config.with_private_key {
+        return Ok(config.account_address.clone());
+    }
+    Ok(equote::get_ethereum_address(eid).or(Err(EnclaveError::UnDetailedEnclaveErr))?.to_hex())
+}
+
+#[logfn(INFO)]
+pub fn start(eid: sgx_enclave_id_t) -> Result<(), ControllerError> {
+    let opt = controller::options::Opt::from_args();
+    let config = KMConfig::load_config(opt.principal_config.as_str())?;
+
+    let mut path = storage_dir(ENCLAVE_DIR).map_err(|_| ControllerError::VerifierError(VerifierError::CreateErr))?;
+    let ethereum_address = get_ethereum_address(eid, config.clone()).map_err(|e| ControllerError::EnclaveError(e))?;
+
+    if opt.sign_address {
+        path.push("principal-sign-addr.txt");
+        let mut file = File::create(&path).map_err(|_| ControllerError::VerifierError(VerifierError::CreateErr))?;
+
+        let signing_address = get_signing_address(eid).map_err(|e| ControllerError::EnclaveError(e))?;
+        let prefixed_signing_address = format!("0x{}", signing_address);
+
+        file.write_all(prefixed_signing_address.as_bytes()).map_err(|_| ControllerError::VerifierError(VerifierError::WriteErr))?;
+        println!("Wrote signing address: {:?} in file: {:?}", prefixed_signing_address, path);
+
+        path.pop();
+        path.push("ethereum-account-addr.txt");
+        let mut file = File::create(&path).map_err(|_| ControllerError::VerifierError(VerifierError::CreateErr))?;
+
+        let prefixed_ethereum_address = format!("0x{}", ethereum_address);
+
+        file.write_all(prefixed_ethereum_address.as_bytes()).map_err(|_| ControllerError::VerifierError(VerifierError::WriteErr))?;
+        println!("Wrote ethereum address: {:?} in file: {:?}", prefixed_ethereum_address, path);
+
+    } else {
+        let private_key = config.private_key.from_hex().map_err(|_| ConfigError::Parsing)?;
+        let ethereum_signer = create_signer(eid, config.with_private_key, &private_key);
+
+        let contract_address = opt.contract_address.unwrap_or_else(|| config.enigma_contract_address.clone());
+        let enigma_contract = EnigmaContract::from_deployed(
+            &contract_address,
+            Path::new(&config.enigma_contract_path),
+            Some(&ethereum_address),
+            config.chain_id,
+            &config.url,
+            ethereum_signer,
+        ).map_err(|e| ControllerError::GenericError(e))?;
```

Comment:
In any context where you pass a closure of the form `|x| Wrapper(x)` (whether `Wrapper` is a tuple struct or an enum tuple variant with one field) you can use the name of `Wrapper` as a function that does the same. Therefore here you can do the following:
```suggestion
        ).map_err(ControllerError::GenericError)?;
```

Reply:
Same story with `ControllerError::VerifierError` below